mesonbuild/compilers/mixins/tasking.py

```diff
@@ -0,0 +1,117 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2012-2023 The Meson development team
+from __future__ import annotations
+
+"""Representations specific to the TASKING embedded C/C++ compiler family."""
+
+import os
+import typing as T
+
+from ...mesonlib import EnvironmentException, OptionKey
+
+if T.TYPE_CHECKING:
+    from ...compilers.compilers import Compiler
+else:
+    # This is a bit clever, for mypy we pretend that these mixins descend from
+    # Compiler, so we get all of the methods and attributes defined for us, but
+    # for runtime we make them descend from object (which all classes normally
+    # do). This gives up DRYer type checking, with no runtime impact
+    Compiler = object
+
+tasking_buildtype_args: T.Dict[str, T.List[str]] = {
```

Comment:
Could you use `T.Mapping` instead of `T.Dict` for these? Mapping doesn't support assignment after iniitalization so Mypy will catch and warn us that someone has modified these, which we don't expect